sitemedia/js/controllers/ittpanel_controller.js

```diff
@@ -106,39 +87,86 @@ export default class extends Controller {
-        const edTop = this.transcriptionTarget
-            .querySelector("ol")
-            .getBoundingClientRect().top;
-        const trTop = this.translationTarget
-            .querySelector("ol")
-            .getBoundingClientRect().top;
-        if (edTop < trTop) {
-            this.transcriptionTarget.querySelector("ol").style.paddingTop = `${
-                trTop - edTop
-            }px`;
-        } else if (trTop < edTop) {
-            this.translationTarget.querySelector("ol").style.paddingTop = `${
-                edTop - trTop
-            }px`;
+        // loop through each transcription and translation (only as many as needed)
+        const minTargets = Math.min(
+            this.transcriptionTargets.length,
+            this.translationTargets.length
+        );
+        for (let i = 0; i < minTargets; i++) {
+            // loop through as many OLs in each transcription/translation as needed
+            const edOls = this.transcriptionTargets[i].querySelectorAll("ol");
+            const trOls = this.translationTargets[i].querySelectorAll("ol");
+            const minLists = Math.min(edOls.length, trOls.length);
+            for (let j = 0; j < minLists; j++) {
+                // first, align tops of lists (using inline styles)
+                const edOl = edOls[j];
+                const trOl = trOls[j];
+                const edTop = edOl.getBoundingClientRect().top;
+                // translation is always 1 pixel difference
+                const trTop = trOl.getBoundingClientRect().top - 1;
+                if (edTop < trTop) {
+                    edOl.style.paddingTop = `${trTop - edTop}px`;
+                } else if (trTop < edTop) {
+                    trOl.style.paddingTop = `${edTop - trTop}px`;
+                }
+                // then, align each line of transcription to translation
+                const edLines = edOl.querySelectorAll("li");
+                const trLines = trOl.querySelectorAll("li");
+                // only align as many lines as we need to
+                const minLines = Math.min(edLines.length, trLines.length);
+                for (let k = 0; k < minLines; k++) {
+                    if (edLines[k] && trLines[k]) {
+                        // calculate number of lines based on line height
+                        const maxLineCount = Math.max(
+                            this.getLineCount(edLines[k]),
+                            this.getLineCount(trLines[k])
+                        );
+                        // set data-lines attribute on each li according to which is longer
+                        edLines[k].setAttribute("data-lines", maxLineCount);
+                        trLines[k].setAttribute("data-lines", maxLineCount);
+                    }
+                }
+            }
+        }
+    }
+
+    removeAlignment() {
+        // remove alignment
+        // first remove data-lines from each line
+        if (this.hasTranscriptionTarget) {
```

Comment:
seems like some redundancy, wondering if there's a nice way to iterate over all like-elements; but that's a nice-to-have, not essential